src/main/java/com/cedarsoftware/util/io/WriteOptionsBuilder.java

```diff
@@ -0,0 +1,817 @@
+            if (customWriter == null) {
+                throw new JsonIoException("Note: class not found (custom JsonClassWriter class): " + writerClassName + ", listed in resources/customWriters.txt as a custom writer for: " + className);
+            }
+            try {
+                JsonWriter.JsonClassWriter writer = (JsonWriter.JsonClassWriter) MetaUtils.newInstance(customWriter, null);
+                writers.put(clazz, writer);
+            } catch (Exception e) {
+                throw new JsonIoException("Note: class failed to instantiate (a custom JsonClassWriter class): " + writerClassName + ", listed in resources/customWriters.txt as a custom writer for: " + className);
+            }
+        }
+        return writers;
+    }
+
+    /**
+     * Load custom writer classes based on contents of resources/customWriters.txt.
+     * Verify that classes listed are indeed valid classes loaded in the JVM.
+     *
+     * @return Map<Class < ?>, JsonWriter.JsonClassWriter> containing the resolved Class -> JsonClassWriter instance.
+     */
+    private static Map<Class<?>, Map<String, String>> loadNonStandardMethodNames() {
+        Map<String, String> map = MetaUtils.loadMapDefinition("nonStandardAccessors.txt");
+        Map<Class<?>, Map<String, String>> nonStandardMapping = new ConcurrentHashMap<>();
+        ClassLoader classLoader = WriteOptions.class.getClassLoader();
+
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            String className = entry.getKey();
+            String mappings = entry.getValue();
+            Class<?> clazz = MetaUtils.classForName(className, classLoader);
+            if (clazz == null) {
+                System.out.println("Class: " + className + " not defined in the JVM");
+                continue;
+            }
+
+            Map<String, String> mapping = nonStandardMapping.computeIfAbsent(clazz, c -> new ConcurrentHashMap<>());
+            for (String split : mappings.split(",")) {
+                String[] parts = split.split(":");
+                mapping.put(parts[0].trim(), parts[1].trim());
+            }
+        }
+        return nonStandardMapping;
+    }
+
+
+    /**
+     * Load the list of classes that are intended to be treated as non-referenceable, immutable classes.
+     *
+     * @return Set<Class < ?>> which is the loaded from resource/nonRefs.txt and verified to exist in JVM.
+     */
+    static Set<Class<?>> loadNonRefs() {
+        Set<Class<?>> nonRefs = new LinkedHashSet<>();
+        Set<String> set = MetaUtils.loadSetDefinition("nonRefs.txt");
+        set.forEach((className) -> {
+            Class<?> clazz = MetaUtils.classForName(className, WriteOptions.class.getClassLoader());
+            if (clazz == null) {
+                throw new JsonIoException("Class: " + className + " undefined.  Cannot be used as non-referenceable class, listed in resources/nonRefs.txt");
```

Comment:
Perhaps this should just be a System.out instead of an exception throw?  The question to ask, is should they still be able to run if they leave an old class in here that is nonReferenceable.  Probably don't need to stop them, just casually let them know via sout.